notebooks/Notebook3.md

```diff
@@ -2,33 +2,86 @@
+
+* Adding grabbable items to the player's inventory
+* Using items on other items (leading to consequences and the item leaving your inventory)
+* Supporting items that contain other items, like a chest
+
+leaving features like locked doors and characters giving items on certain lines of dialogue for next week. I also felt that getting the listed things working would make what remained a lot easier.
+
+I started by producing some DSL lines that would include all these new features. It includes a key in one room that unlocks a chest containing gold in another room:
+
+```
+<dark>(treasure is north, cave is south) {
+    You find yourself in a dark room. A door leads to the north, while a cave opens to the south. A mysterious locked door also lies to the west.
+
+    <beggar>(character) {
+        apple: Yes, those are my apples.
+        apples: Yes, those are my apples.
+        bucket: Do you like my bucket?
+        door: Go through that door if you dare
+    }
+
+    <key>(item, grabbable, opens chest) {
+        look: It's definitely a key.
+        pickup: You've grabbed a key and slipped it into your pocket.
+        use: With a snap, the key slides into place and unlocks the chest.
+    }
+}
+
+<cave>(dark is north) {
+    You've arrived in a dingy cave. You can go north to return to the starting room.
+}
+
+<treasure>() {
+    Wow, there's a treasure chest in here! You can't open it though (at least not yet). The starting room resides to the south.
+
+    <chest>(item, contains gold) {
+        look: It's a chest.
+    }
+    <gold>(item, grabbable) {
+        look: It's very shiny!
+        pickup: Congratulations, you have found the magic treasure that makes you win the game!
+    }
+}
+```
+
+From there, I wrote all the parsing code (using both the pyparsing package and my own code from scratch) such that all the info laid out in the DSL is imported into the backend and the game can run. As of right now, all the items are properly assigned to each room, and you're able to 'look at' them and it will print out the proper text. I still need to work on the backend to support the inventory system and item interactions. So not as much has been done as I hoped this week, but I have a few work hours on Monday before studio time, and I'll update this notebook if anything meaningful gets accomplished then!
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+The most pressing issue is finishing up the backend of my current items workload and implementing what was left out my items list for this week. Once the implementation is done, there is still a good amount of error-handling and testing to be done to make sure that my code is robust and intuitive to use.
```

Comment:
Error handling is definitely a great goal to work towards. I remember you saying that you had to kind of wrangle the Python parser code to do what you wanted it to, though, so hopefully error handling isn't as much of a hassle. To save time, I'd probably just focus on the errors that people are most likely to make (so prioritization).